Mlem/Views/Tabs/Posts/Components/Shared/Links/User Profile Link.swift

```diff
@@ -26,24 +43,117 @@ struct UserProfileLink: View
                     }
                 }
                 
-                Text(user.name)
+                /* Disabled icons for now */
+                if let flair = calculateFlair() {
+                    if let flairSystemIcon = flair.systemIcon {
+                        Image(systemName: flairSystemIcon).foregroundColor(flair.color)
+                    }
+                }
+                
+                
+                // User display name if one exists
+                Text(user.displayName ?? user.name)
                     .minimumScaleFactor(0.01)
                     .lineLimit(1)
             }
-            .if(user.admin)
+            .if(calculateFlair() != nil)
             { viewProxy in
                 viewProxy
-                    .foregroundColor(.red)
+                    .foregroundColor(calculateFlair()!.color).bold()
             }
-            .if(user.botAccount == true)
-            { viewProxy in
-                viewProxy
-                    .foregroundColor(.indigo)
+        }
+    }
+    
+    struct UserProfileLinkFlair {
+        var color: Color
+        var systemIcon: String? = nil
+    }
+    
+    private func calculateFlair() -> UserProfileLinkFlair? {
+        if let userServer = user.actorId.host() {
+            if UserProfileLink.developerNames.contains(where: { $0 == "\(userServer)\(user.actorId.path())" }) {
+                return UserProfileLink.flairDeveloper 
             }
-            .if(user.name == "lFenix")
-            { viewProxy in
-                viewProxy
-                    .foregroundColor(.yellow)
+        }
+        if user.admin {
+            return UserProfileLink.flairAdmin
+        }
+        if user.botAccount {
+            return UserProfileLink.flairBot
+        }
+        if commentContext != nil {
+            if commentContext!.distinguished {
+                return UserProfileLink.flairMod
+            }
+        }
+        if postContext != nil {
+            if user == postContext!.creator {
+                return UserProfileLink.flairOP
+            }
+        }
+        return nil
+    }
+}
+
+// TODO: darknavi - Move these to a common area for reuse
```

Comment:
Apple did introduced a common area for this, and they're then stripped from the bundle for production too 🙌

https://www.avanderlee.com/xcode/development-assets-preview-catalog/

It's worth a discussion amongst the team as to how we approach providing mock data for more complex previews, we might want to consider having blobs of JSON in the development assets for example, and then similar to the linked article we could extend the types we care about with a `.preview` property or such to pass in to the preview area?

Reply:
JSON blobs sounds good to me, although some code-generated stuff would be useful to quickly generating lots of cases (different types of posts, etc.)

I'll poke around the dev assets catalog when I get a chance!